src/ex4/server/services/storage.js

```diff
@@ -0,0 +1,72 @@
+import Item from "../db/models/item.js";
+import { sequelize } from "../db/models/index.js";
+
+class StorageService {
+  async getTasks() {
+    return await Item.findAll({
+      attributes: [
+        "id",
+        "itemName",
+        "checked",
+        [
+          sequelize.fn(
+            "DATE_FORMAT",
+            sequelize.col("doneTime"),
+            "%Y-%m-%d %H:%i:%s"
+          ),
+          "doneTime",
+        ],
+      ],
+      raw: true,
+    });
+  }
+
+  async addTask(itemName) {
+    const addedTask = await Item.create({ itemName }, { raw: true });
+    return addedTask.toJSON().id;
+  }
+
+  async deleteTask(id) {
+    return await Item.destroy({ where: { id } });
+  }
+
+  async clearAll() {
+    return await Item.destroy({ truncate: true });
```

Comment:
a common solution is to have another field in the table `deleted` and marked it as true, as you don't want to lose data or let the user the ability to restore data.
it will require from you to add new field and fetch items that is `deleted: false`.